requires.py

```diff
@@ -21,23 +21,47 @@ class EtcdClient(RelationBase):
 
     @hook('{requires:etcd}-relation-{joined,changed}')
     def changed(self):
+        ''' Indicate the relation is connected, and if the relation data is
+        set it is also available. '''
         self.set_state('{relation_name}.connected')
+
         if self.connection_string():
-            self.set_state('{relation_name}.available')
+            cert = ssl_certificates()
+            if cert['client_cert'] and cert['client_key'] and cert['client_ca']:  # noqa
+                self.set_state('{relation_name}.tls.available')
+            else
+              self.set_state('{relation_name}.available')
 
     @hook('{requires:etcd}-relation-{broken, departed}')
     def broken(self):
+        ''' Indicate the relation is no longer available and not connected. '''
         self.remove_state('{relation_name}.available')
         self.remove_state('{relation_name}.connected')
 
-    def connection_string(self):
-        '''
-        Get the connection string, if available, or None.
-        '''
+    def get_connection_string(self):
+        ''' Return the connection string, if available, or None. '''
         return self.get_remote('connection_string')
 
-    def ssl_certificates(self):
-        ''' Return a dict with the client key and certificate '''
+    def get_client_credentials(self):
+        ''' Return a dict with the client certificate, ca and key to
+        communicate with etcd using tls. '''
         return {'client_cert': self.get_remote('client_cert'),
                 'client_key': self.get_remote('client_key'),
                 'client_ca': self.get_remote('client_ca')}
+
+    def save_client_credentials(self, key, cert, ca):
```

Comment:
so we call this method to put bits on disk?

like:

save_client_credentials('/etc/ssl/unit_name/cert.pem', '/etc/ssl/unit_name/key.pem', '/etc/ssl/unit_name/ca.pem')
?